src/Rules/Actions.hs

```diff
@@ -126,6 +128,18 @@ copyDirectory source target = do
     putProgressInfo $ renderAction "Copy directory" source target
     quietly $ cmd cmdEcho ["cp", "-r", source, target]
 
+-- | Copy the content of the source directory into the target directory. Only
+-- the files and directories for which the predicate returns True are copied.
+copyDirectoryContent :: (FilePath -> IO Bool) -> FilePath -> FilePath -> Action ()
```

Comment:
On a second thought, perhaps this should be simplified even further? Do we really need to test filenames in `IO` or `Action`? The only (so far) use case in `sdist` rule is just a pure function.